planetiler-benchmarks/src/main/java/com/onthegomap/planetiler/benchmarks/BenchmarkMbtilesWriter.java

```diff
@@ -0,0 +1,254 @@
+package com.onthegomap.planetiler.benchmarks;
+
+import com.onthegomap.planetiler.Profile;
+import com.onthegomap.planetiler.VectorTile;
+import com.onthegomap.planetiler.collection.FeatureGroup;
+import com.onthegomap.planetiler.config.Arguments;
+import com.onthegomap.planetiler.config.MbtilesMetadata;
+import com.onthegomap.planetiler.config.PlanetilerConfig;
+import com.onthegomap.planetiler.geo.GeometryType;
+import com.onthegomap.planetiler.geo.TileCoord;
+import com.onthegomap.planetiler.mbtiles.MbtilesWriter;
+import com.onthegomap.planetiler.render.RenderedFeature;
+import com.onthegomap.planetiler.stats.Counter;
+import com.onthegomap.planetiler.stats.Stats;
+import com.onthegomap.planetiler.stats.Timers;
+import com.onthegomap.planetiler.util.MemoryEstimator.HasEstimate;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.LongSummaryStatistics;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.LongSupplier;
+import java.util.function.Supplier;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BenchmarkMbtilesWriter {
```

Comment:
When I run with one repetition, 10m tiles, and without compact-db I see this output at the end of each stage:

compact db:
```
0:00:15 INF [mbtiles] - Finished in 13s cpu:33s avg:2.6
0:00:15 INF [mbtiles] -   read    1x(4% 0.5s wait:12s)
0:00:15 INF [mbtiles] -   encode 10x(13% 2s wait:10s)
0:00:15 INF [mbtiles] -   write   1x(97% 12s)
```

original:
```
0:00:34 INF [mbtiles] - Finished in 19s cpu:2m38s avg:8.2
0:00:34 INF [mbtiles] -   read    1x(2% 0.3s wait:18s)
0:00:34 INF [mbtiles] -   encode 10x(73% 14s sys:2s block:5s)
0:00:34 INF [mbtiles] -   write   1x(76% 15s wait:4s)
```

but the summary statistics show 19 seconds for all stages on the original, and 13 for all on compact - so they are measuring the longest phase but the main bottleneck we care about here is the write thread.  Would it be possible to have the benchmark isolate just the `write` phase by invoking the compact vs. non-compact batched writer directly? The tile `byte[]` payloads don't need to actually be real tiles.

Also I usually find it a little more straightforward to report the performance in "operations per second" to make it easier to compare across different input sizes.

Reply:
Despite the name `BenchmarkMbtilesWriter`, I wanted to check both stages encode and write. And it's specifically designed to make the current memoization not work.
This is related to https://github.com/onthegomap/planetiler/pull/219#discussion_r871772440
=> if that's the case, then I'll adjust according

Reply:
I think the options would either be to benchmark the encode/write separately, or try to extract the cpu time from each thread since they all end at the same time.  I'm mostly interested in the write thread performance since that is a single threaded bottleneck.

Reply:
addressed in https://github.com/onthegomap/planetiler/pull/219/commits/44d481e74624329f27d86709e3aa86e09c95f35a

I tried to make the the test somewhat realistic... the "key" is the "dupe spread". If I am not mistaken, then the dupe spread for Australia is 0.17%

results are as following:
```
--bench-tiles-to-write=1000000 --bench-repetitions=10 --bench-no-dupe-tiles=10 --bench-distinct-tile-data-size=800 --bench-dupe-tile-data-size=100

  --bench-dupe-spread=10

    (non-compact)
    tileWritesPerSecondsStats: DoubleSummaryStatistics{count=10, sum=5535826.018941, min=494772.881352, average=553582.601894, max=572869.469085}

    --compact-db
    tileWritesPerSecondsStats: DoubleSummaryStatistics{count=10, sum=5893505.249478, min=516823.356896, average=589350.524948, max=614381.936503}

  --bench-dupe-spread=1

    (non-compact)
    tileWritesPerSecondsStats: DoubleSummaryStatistics{count=10, sum=5515072.624620, min=500652.172801, average=551507.262462, max=572924.290469}

    --compact-db
    tileWritesPerSecondsStats: DoubleSummaryStatistics{count=10, sum=6341896.356866, min=553440.630765, average=634189.635687, max=652745.072077}

  --bench-dupe-spread=50

    (non-compact)
    tileWritesPerSecondsStats: DoubleSummaryStatistics{count=10, sum=5517730.613009, min=493311.037536, average=551773.061301, max=578336.895536}

    --compact-db
    tileWritesPerSecondsStats: DoubleSummaryStatistics{count=10, sum=4996218.743876, min=434994.065552, average=499621.874388, max=517147.994776}
```

Reply:
A "dupe spread" of 50 would mean that in the "dupe group" always two tiles would share the same data. At that level the compact mode has disadvantages because of the additional overhead (more inserts). 
The less spread dupes are (0 would mean all tiles in the "dupe group" would share the tile data), the more efficient the compact mode gets.

Reply:
Sounds good, a real-world test on planet data showed about a 5% improvement going to compact, so 10 seems like a realistic default value to use :thumbsup: